File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolForm.ts
import { StringEditor, TextAreaEditor, IntegerEditor, LookupEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SchoolForm {
    Name: StringEditor;
    Description: TextAreaEditor;
    EType: IntegerEditor;
    Address: StringEditor;
    City: StringEditor;
    StateId: LookupEditor;
    DistrictId: LookupEditor;
    TalukaId: LookupEditor;
    LocationInfo: StringEditor;
    EstablishmentDate: DateEditor;
}

export class SchoolForm extends PrefixedContext {
    static readonly formKey = 'Schools.School';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SchoolForm.init)  {
            SchoolForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = IntegerEditor;
            var w3 = LookupEditor;
            var w4 = DateEditor;

            initFormType(SchoolForm, [
                'Name', w0,
                'Description', w1,
                'EType', w2,
                'Address', w0,
                'City', w0,
                'StateId', w3,
                'DistrictId', w3,
                'TalukaId', w3,
                'LocationInfo', w0,
                'EstablishmentDate', w4
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Schools/School/SchoolForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Schools.Forms;

[FormScript("Schools.School")]
[BasedOnRow(typeof(SchoolRow), CheckNames = true)]
public class SchoolForm
{
    [HalfWidth]
    public string Name { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    [HalfWidth]
    public short EType { get; set; }
    [HalfWidth]
    public string Address { get; set; }
    [HalfWidth]
    public string City { get; set; }
    [HalfWidth]
    public int StateId { get; set; }
    [HalfWidth]
    public int DistrictId { get; set; }
    [HalfWidth]
    public int TalukaId { get; set; }
    [HalfWidth]
    public string LocationInfo { get; set; }
    [HalfWidth]
    public DateTime EstablishmentDate { get; set; }
}